### DISCUSSION TITLE
Alternative to RaggedTensors

### DISCUSSION THREAD

**Discussion by System0X** [keras-team/keras#20290] [Q&A]
Hello, 
I know that RaggedTensor were removed from keras 3. Is there an alternative? Or can we just use RaggedTensors with tensorflow backend?
Maybe a linked question but why has RaggedTensors been removed is it better not to use them? Maybe just pad out the result?

**Comment by fchollet** [comment]
They were removed because they would only work with TF. Keras 3 focused on workflows that can be run with any backend.

There is no real alternative -- my recommendation is to keep your usage of ragged tensors confined to your tf.data pipeline. You must densify/pad your tensors before you hit the model.
